kcalcore/recurrencerule.cpp

```diff
@@ -623,6 +623,16 @@ void Constraint::appendDateTime( const QDate &date, const QTime &time,
                                  QList<KDateTime> &list ) const
 {
   KDateTime dt( date, time, timespec );
+  KDateTime nonTimespecDt( date, time, KDateTime::Spec::UTC() );
+  if ( !dt.isValid() && nonTimespecDt.isValid() && timespec.utcOffset() == 0 ) {
+    // This is to fix an issue where CalDAV servers can provide a VTIMEZONE
+    // definition for CEST (eg "Europe/Berlin") with a DAYLIGHT component like:
+    // "DTSTART:19700329T020000\r\n"
+    // "RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3\r\n"
+    // This will fail after 1981 since the system definition differs, resulting
+    // in an invalid KDateTime due to the "invalid" timespec.
```

Comment:
The problem is not the timespec value, but the `QDateTime` value, i.e. `KDateTime.isValid()` is false, because the enclosed `QDateTime` value is not valid.
The `KDateTime` is initialized with a `QDateTime` having `Qt::LocalTime` set as its timezone information. (https://github.com/mer-packages/kcalcore/blob/master/kcalcore/kdedate/kdatetime.cpp#L809)

Therefore, the timestamp that marks the beginning of the DST period is invalid, cf. also
http://doc.qt.io/qt-5/qdatetime.html#isValid

```
qWarning() << QDateTime( QDate( 2015, 03, 29 ), QTime( 02, 00, 00 ), Qt::LocalTime).isValid();
```
is false if your system time is set to 'Europe/Berlin'.
This applies for all timezones that observe DST, e.g. the following is false if the system time is set to Australia/Sydney.

```
qWarning() << QDateTime( QDate( 2015, 10, 04 ), QTime( 02, 00, 00 ), Qt::LocalTime).isValid();
```

Unfortunately, I have currently no idea how this could be fixed. My bet would be to use UTC based times to calculate these transitions, but this is only a guess due to the lack of alternatives...